osh/word_eval.py

```diff
@@ -798,24 +800,30 @@ def _EvalVarRef(self, val, blame_tok, quoted, vsub_state, vtest_place):
         Args:
           blame_tok: 'foo' for ${!foo}
         """
+        probe('word_eval', '_EvalVarRef_enter')
         UP_val = val
         with tagswitch(val) as case:
             if case(value_e.Undef):
+                probe('word_eval', '_EvalVarRef_exit')
                 return value.Undef  # ${!undef} is just weird bash behavior
 
             elif case(value_e.Str):
                 val = cast(value.Str, UP_val)
                 bvs_part = self.unsafe_arith.ParseVarRef(val.s, blame_tok)
+                probe('word_eval', '_EvalVarRef_exit')
                 return self._VarRefValue(bvs_part, quoted, vsub_state,
                                          vtest_place)
 
             elif case(value_e.BashArray):  # caught earlier but OK
+                probe('word_eval', '_EvalVarRef_exit')
                 e_die('Indirect expansion of array')
 
             elif case(value_e.BashAssoc):  # caught earlier but OK
+                probe('word_eval', '_EvalVarRef_exit')
                 e_die('Indirect expansion of assoc array')
```

Comment:
not sure how precious we should be about capturing exits before e_die()and assertions... thoughts?